### PR TITLE
Issue 1865: Add a rule using the maven-enforcer plugin that requires JDK 8

### PR DESCRIPTION
**Pull request by AricMonary**
# Pull Request Description

This pull request closes strongbox/strongbox#1865

# Acceptance Test

* [x] Building the code of the [`strongbox-parent`](https://github.com/strongbox/strongbox-parent) with `mvn clean install` still works.
* [x] Building the code of the [`strongbox`](https://github.com/strongbox/strongbox) project with `mvn clean install -Dintegration.tests` still works.
* [x] Running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [x] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [x] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Questions

* Does this pull request break backward compatibility? 
  * [ ] Yes
  * [X] No

* Does this pull request require other pull requests to be merged first? 
  * [ ] Yes, please see #...
  * [x] No

* Does this require an update of the documentation?
  * [ ] Yes, please see strongbox/strongbox-docs#{PR_NUMBER}
  * [x] No

# Code Review And Pre-Merge Checklist

* [x] My code follows the [coding convention](https://strongbox.github.io/developer-guide/coding-convention.html) of this project.
* [x] I have performed a self-review of my own code.
* [x] I have commented my code in hard-to-understand areas.
* [x] My changes generate no new warnings.

### CODE REVIEW

**Comment by AricMonary** [comment]
There is a limitation with activating profiles for specific versions of Java. Although you can support multiple ranges it only appears to look at the first range, meaning that the three out of bounds ranges ( ,jdk8), [jdk9,jdk11), and [jdk12, ) must be separated. May be bypassed through the use of properties.

File: pom.xml [hunk 1]
```diff
@@ -1641,6 +1640,160 @@
...
+                            <execution>
+                                <id>enforce-versions</id>
+                                <goals>
+                                    <goal>enforce</goal>
+                                </goals>
+                                <configuration>
+                                    <rules>
+                                        <requireJavaVersion>
+                                            <version>[11.3,)</version>
+                                            <message>JDK 11 is not currently supported. Strongbox will only support JDK 11.3, or higher, in the future.</message>
```
Comment (carlspring) [inline pom.xml:1663]:
```suggestion
                                            <message>JDK 11 is not currently supported. Strongbox will only support JDK 11.3, or higher.</message>
```

**Comment by AricMonary** [comment]
I will confirm that I did test this with JDK7, JDK8, JDK11 (Version that was less than 11.3), JDK11.3, and JDK12.

File: pom.xml [hunk 2]
```diff
@@ -419,9 +419,22 @@
...
+                        <configuration>
+                            <rules>
+                                <requireJavaVersion>
+                                     <version>[1.8.0-241,1.9)</version>
+                                    <message>Invalid JDK version, minimum required is 1.8.0_241.</message>
+                                </requireJavaVersion>
+                            </rules>
+                        </configuration>
+                    </execution>
+				</executions>
```
Comment (carlspring) [inline pom.xml:436]:
```suggestion
                </executions>
```

File: pom.xml [hunk 3]
```diff
@@ -419,9 +419,22 @@
...
+                        </goals>
+                        <configuration>
+                            <rules>
+                                <requireJavaVersion>
+                                     <version>[1.8.0-241,1.9)</version>
+                                    <message>Invalid JDK version, minimum required is 1.8.0_241.</message>
+                                </requireJavaVersion>
+                            </rules>
+                        </configuration>
+                    </execution>
```
Comment (carlspring) [inline pom.xml:435]: Could you please wrap this up in a Maven profile that is only activated when the JDK is 8? Thanks! :)

Comment (carlspring) [reply pom.xml:435]: Actually, my bad, you could have multiple versions defined.

**Comment by steve-todorov** [comment]
I've slightly changed the pom so that it allows for both JDK8 and JDK11, but still fails for all other JDK versions. :)
Expected failure outputs are:

* JDK != 8 || 11	
    ```
    [ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M3:enforce (enforce-jdk-and-maven-versions) on project strongbox-parent: org.apache.maven.plugins.enforcer.RequireJavaVersion failed with message:
    [ERROR] 
    [ERROR]  AdoptOpenJDK 13b13 is not a supported JDK version! 
    [ERROR]  See https://strongbox.github.io/developer-guide/getting-started.html 
    [ERROR] 
    [ERROR] -> [Help 1]
    ```

* MVN Version < 3.6.3:
    ```
    [ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M3:enforce (enforce-jdk-and-maven-versions) on project strongbox-parent: org.apache.maven.plugins.enforcer.RequireMavenVersion failed with message:
    [ERROR] 
    [ERROR]  You are using an unsupported Maven version! 
    [ERROR]  See https://strongbox.github.io/developer-guide/getting-started.html 
    [ERROR] 
    [ERROR] -> [Help 1]
    ```

Thanks @AricMonary for your initial work on this! :) 

//cc @carlspring if you have no more comments you can proceed with merging :)

**Pull request merged**